### DISCUSSION TITLE
does rqlite support recursive CTE?

### DISCUSSION THREAD

**Discussion by l1t1** [rqlite/rqlite#1729] [Q&A] [answered]
` with recursive t(a) as (select 1 as a union all select 1+a from t where a<10000) select count(*) from t;`
says
`1 row affected`, but didn't return `10000`

**Comment by l1t1** [comment]
it can insert into a table,but cannot select the result?
```
INSERT INTO foo(name) with recursive t(a) as (select 1 as a union all select 1+a from t where a<100) select count(*
1 row affected

127.0.0.1:4001> SELECT * FROM foo
+----+-------+
| id | name  |
+----+-------+
| 1  | fiona |
+----+-------+
| 2  | 100   |
```

**Comment by otoolep** [comment]
The rqlite CLI is pretty simple, and just checks if a SQL statement starts with `SELECT` to determine whether it is a read or write.

For more complex SQL statements, use the API directly. See https://rqlite.io/docs/api/api/

**Comment by otoolep** [comment, marked as answer]
` with recursive t(a) as (select 1 as a union all select 1+a from t where a<10000) select count(*) from t;`

Send this to the `/db/query` endpoint of rqlite, it should work.
